autocertkit/utils.py

```diff
@@ -2370,6 +2371,37 @@ def set_nic_device_status(session, interface, status):
     wait_for_linkstate(session, interface, status)
     time.sleep(5)
 
+@log_exceptions
+def get_eth_by_bridge(session, bridge: str) -> str:
+    """
+    Get the device name given a bridge name.
+    """
+    host = session.xenapi.session.get_this_host(session.handle)
+    pifs = session.xenapi.host.get_PIFs(host)
+    for network in session.xenapi.network.get_all():
+        if session.xenapi.network.get_bridge(network) == bridge:
+            for pif in pifs:
```

Comment:
possible to ignore non physical pif by `session.xenapi.PIF.get_physical(pif)`, refer to https://github.com/xenserver/auto-cert-kit/blob/master/plugins/autocertkit#L1889